core/src/main/java/dev/streamx/cli/ingestion/BasePublishUnpublishCommand.java

```diff
@@ -0,0 +1,74 @@
+package dev.streamx.cli.ingestion;
+
+import dev.streamx.cli.exception.IngestionClientException;
+import dev.streamx.clients.ingestion.StreamxClient;
+import dev.streamx.clients.ingestion.exceptions.StreamxClientException;
+import jakarta.inject.Inject;
+import java.util.List;
+import org.apache.commons.lang3.StringUtils;
+import picocli.CommandLine.ArgGroup;
+import picocli.CommandLine.Model.CommandSpec;
+import picocli.CommandLine.Spec;
+
+public abstract class BasePublishUnpublishCommand implements Runnable {
+
+  @ArgGroup(exclusive = false)
+  protected IngestionArguments ingestionArguments;
+
+  @Spec
+  protected CommandSpec spec;
+
+  @Inject
+  protected SchemaProvider schemaProvider;
+
+  @Inject
+  protected StreamxClientProvider streamxClientProvider;
+
+  protected abstract void performCommand(StreamxClient client) throws StreamxClientException;
+
+  @Override
+  public void run() {
+    StreamxClient client = createClient();
+
+    try {
+      performCommand(client);
+    } catch (StreamxClientException e) {
+      if (is404Error(e)) {
+        createExtendedIngestionClientException(e);
+      } else {
+        createIngestionClientException(e);
+      }
+    } finally {
+      try {
+        client.close();
+      } catch (StreamxClientException e) {
+        throw new IngestionClientException(e);
+      }
+    }
+  }
+
+  private StreamxClient createClient() {
+    try {
+      return streamxClientProvider.createStreamxClient();
+    } catch (StreamxClientException e) {
+      throw new IngestionClientException(e);
+    }
+  }
+
+  private boolean is404Error(StreamxClientException e) {
+    // TODO: extend StreamxClientException class to contain int httpStatusCode field
+    //  and use it here instead of analyzing exception message
+    return StringUtils.contains(e.getMessage(), "Response status: 404");
+  }
```

Comment:
TODO for the future.
For now the error message is stable..

Reply:
avoid todo because it tends to be forgotten. If nothing blocks u now please implement. otherwise pleas add a jira ticket

Reply:
This TODO is a sign that more changes are required.
In
 - streamx cli
 - streamx 
 - ingestion-service

Therefore this PR will not be merged yet